actors/builtin/miner/miner_state_test.go

```diff
@@ -440,6 +442,252 @@ func TestVestingFundsStore(t *testing.T) {
 	assert.Equal(t, abi.NewTokenAmount(51), vested)
 }
 
+// TODO make top level tests for vested, unvested and then together.
+
+func TestVestingFunds(t *testing.T) {
+	t.Run("Vest all funds in single epoch", func(t *testing.T) {
+		harness := constructStateHarness(t, abi.ChainEpoch(0))
+		vspec := &miner.VestSpec{
+			InitialDelay: 0,
+			VestPeriod:   1,
+			StepDuration: 1,
+			Quantization: 1,
+		}
+		vestStart := abi.ChainEpoch(10)
+		vestSum := abi.NewTokenAmount(100)
+		harness.addLockedFunds(vestStart, vestSum, vspec)
+		amountVested := harness.unlockVestedFunds(vestStart + 2)
+		assert.Equal(t, vestSum, amountVested)
+		assert.Zero(t, harness.s.LockedFunds.Int64())
+		assert.True(t, harness.vestingFundsStoreEmpty())
+		// subsequent calls to unlock return 0
+		amountVested = harness.unlockVestedFunds(vestStart + 2)
+		assert.Zero(t, amountVested)
+
+	})
+	t.Run("Sum of all entries in vesting store is equal to vested sum", func(t *testing.T) {
```

Comment:
Whoah, this goes from really simple to eye-watering complicated very quickly. Please try to write some tests between the first one and this one, each one of which changes _one_ thing about the schedule:
- vest all at once after non-zero cliff
- period = 2, step = 1,  Q=1 -> [50, 50]
- period = 2, step = 1, Q=2 ->  [0, 100]
- period = 3 , step= 1,  Q=1 -> [33, 33, 34] ???
- period = 3 , step= 1,  Q=2 -> [0, 66, 0, 34] ???
- period = 2, step = 2,  Q=1 -> [0, 100]
- period = 5, step = 2, Q=1 -> [0, 40, 0, 40, 0, 20] ???
- period = 5, step = 2, Q=2 -> [0, 40, 0, 40, 0, 20] ???
- period = 5, step = 3, Q=1 -> [0, 0, 60, 0, 0, 40] ???